### PR TITLE
added a confirmation modal for changes to the running phase and a sho…

### LINKED ISSUE #201: provide user with information about action to change processes
Perhaps we should require confirmation when a user is changing an actively running phase. This also gives us the opportunity to educate them on what is actually happening: "Hey, you are about to change an actively running phase. This will create a new phase immediately there-after and transition it to the active phase (maybe we show them a graphic or something?). I can see how, from a user perspective, the behavior may look like a bug without us explaining it

### PR DESCRIPTION
**Pull request by vdiezel**
…rt explanation for the user

closes #201

### CODE REVIEW

File: src/components/processviewer/body/builder/ChangesFooter.jsx [hunk 1]
```diff
@@ -18,32 +20,39 @@ const mapDispatchToProps = dispatch => ({
...
+    props.setProcessBuilderValidationErrors(errors)
 
-const ChangesFooter = props  => {
+    if (errors.length) return
+
+    const runningPhaseId = getRunningPhase(activeProcess).phaseId.toString()
+    const insertPhase = PDVC.hasStreamTargetChange(diffs, runningPhaseId)
 
-  const { activeProcess, processDescriptionSnapshots } = props
+    if (insertPhase) {
```
Comment (vdiezel) [inline src/components/processviewer/body/builder/ChangesFooter.jsx]: logi where we split between immediate submit and showing the confirmation modal first

File: src/components/processviewer/body/builder/PhaseTabContent.jsx [hunk 2]
```diff
@@ -121,6 +121,12 @@ class PhaseTabContent extends React.Component {
             </Fragment>
           }
         </Form>
+        {this.props.activeProcess.state === 'running' ?
```
Comment (vdiezel) [inline src/components/processviewer/body/builder/PhaseTabContent.jsx]: moved Changesfotter to the top of the view

File: src/utils/PDVC.js [hunk 3]
```diff
@@ -87,8 +87,8 @@ class PDVC {
     return diffs.some(diff => diff.phaseId === runningPhaseId)
   }
 
-  static hasStreamTargetChange(diffs){
-    return diffs.some(diff => diff.type === 'streamTargetChange')
+  static hasStreamTargetChange(diffs, phaseId){
```
Comment (vdiezel) [inline src/utils/PDVC.js]: This was a bug - I wanted to check if a certain phase has streamTargetChanges (for now, we are mainhly interested in the running phase). This caused the insertion of a phase on ad hoc changes when we did modify the runningPhase, but not in a way that needed the insertion of a phase (e.g. chaging the transition method) AND changed the stream target of a future phase.

File: src/components/processviewer/body/builder/ChangesFooter.jsx [hunk 4]
```diff
@@ -18,32 +20,39 @@ const mapDispatchToProps = dispatch => ({
...
+      triggerConfirmationModal(true)
+      return
+    }
+    else await confirmChanges(diffs)
```
Comment (DanielSeehausen) [inline src/components/processviewer/body/builder/ChangesFooter.jsx]: no need for else here, as its always

Comment (vdiezel) [reply src/components/processviewer/body/builder/ChangesFooter.jsx]: :man_facepalming:

**Pull request merged**